src/jdk.compiler/share/classes/com/sun/tools/javac/comp/LambdaToMethod.java

```diff
@@ -879,8 +879,8 @@ private JCExpression makeMetafactoryIndyCall(JCFunctionalExpression tree,
                 }
             }
             if (isQuotable) {
-                VarSymbol reflectField = (VarSymbol)tree.codeModel;
-                staticArgs = staticArgs.append(reflectField.asMethodHandle(true));
+                MethodSymbol opMethodSym = (MethodSymbol)tree.codeModel;
```

Comment:
Since  now all code models are stored as `MethodSymbol`, I wonder if we could maybe sharpen the type in `JCLambda` to use `MethodSymbol` instead of `Symbol`.